lib/inputs/text-input.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+var _ = require('underscore');
+var BaseInput = require('./base-input');
+var errors = require('../errors');
+
+const DEFAULT_VALUE = '';
+
+class TextInput extends BaseInput {
+    constructor(id, params) {
+        super('text', id, params);
+    }
+
+    getInputDefaultValue() {
+        return DEFAULT_VALUE;
+    }
+
+    normalizeParams(params) {
+        if (params.default) {
+            try {
+                params = Object.assign({}, params, {
+                    default: this.normalizeValue(params.default)
+                });
+            } catch (err) {
```

Comment:
Putting the `try`/`catch` around the whole `Object.assign` thing makes if non-obvious from where you expect the error to come from. I’d just do `let default = this.normalizeValue(params.default)`, wrap it into `try`/`catch`, and then do `Object.assign`.